diagnostic.md

```diff
@@ -2,37 +2,43 @@
 
 What are three advantages of using Ember?
 
-```sh
-# your answer here
+```md
+Dynamic Data Binding
+Client Side Casching
+URLs!
 ```
 
 What is the templating system used in Ember? How it is related to
 handlebars?
 
-```sh
-# your answer here
+```md
+They use handlebars within their view framework. That is to say, each view has its own template which may be compiled along with many componants which in turn may have their own templates. These templates are all pumped into a basic ember html boilerplate and may also have extensions with in their HB templates.
 ```
 
 Name at least two layers of an Ember application (1.0 or 2.0):
 
-```sh
-# your answer here
+```md
+View layer
+Routes layer
 ```
 
 How does a component differ from an Ember template?
 
-```sh
-# your answer here
+```md
+componants are like little snippits of code which can have their own compartmentalized javascipt. They can be used within templates but cant be routed to like templates
 ```
 
 What is an opinionated framework? Is Ember opinionated? How or how not?
 
-```sh
-# your answer here
+```md
+It means the language is very particular about layout and semantics. From what I gather ember is very opinionated.
+
 ```
 
 In Ember what is a route's job (please list three)?
 
-```sh
-# your answer here
+```md
+To point to the right view
+To isolate the content being viewed/utilized
+To make make URLs useful
```

Comment:
You're thinking it about the right way, but I can't agree entirely with your phrasing. Does this make sense?

1. Parse information from the url
2. Link the route to a component or template
3. Load the template/UI data via the 'model' hook.